### PR TITLE
Update plot_section to use tight_layout()

### PR DESCRIPTION
**Pull request by Agent6-6-6**
Fixes legend for plotting section going outside plot region

Before:-
before

After:-
after

### CODE REVIEW

**Comment by robbievanleeuwen** [comment]
Thanks @Agent6-6-6, *concreteproperties* also uses the ``plotting_context`` similar to *sectionproperties*. Would the fix be similar to how you solved it with *sectionproperties* [here](https://github.com/robbievanleeuwen/section-properties/pull/235)?

``plotting_context`` in *concreteproperties* is in ``post.py``.

**Comment by Agent6-6-6** [comment]
The difference is all plots in section-properties have the aspect ratio applied, whereas in concrete properties only the section plot requires it.

There is something unique about requiring the aspect ratio to be applied immediately after `tight_layout` to get it to work nicely. Using the `plotting_context` I just couldn't get it to work. But that seems to be the issue if you can figure out a way to get the aspect ratio to work (potentially feeding another true/false variable from plot section and moving the aspect ratio to the `plotting context` might do it?  ..... I didn't try that way to be honest, was heading off on another tangent which just couldn't get to work, only just thought of doing it this way. But you'd maybe have to update more locations where `plotting_context` was utilised for the additional variable.

**Comment by Agent6-6-6** [comment]
Also to prevent issues with the title also it seems to be required to be in this order
- apply title
- apply tight layout
- apply aspect ratio

The `tight_layout` in `plotting_context` did not seem to be applied in the right order

**Comment by Agent6-6-6** [comment]
@robbievanleeuwen updated plotting_context to include the aspect ratio being set. Seems to work fine now at my end. 

Not sure if there are any other plot types where you might need to plot to a equal aspect ratio with the `aspect=True` key word though? Because it is false by default only the section plot function needs it set to True.

**Comment by Agent6-6-6** [comment]
@robbievanleeuwen - Updated two further references where aspect ratio was being set to equal for plotting mesh and stresses.

File: concreteproperties/post.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def plotting_context(
     ax: Optional[matplotlib.axes.Axes] = None,  # type: ignore
     pause: bool = True,
     title: str = "",
+    aspect: bool = False,
```
Comment (robbievanleeuwen) [inline concreteproperties/post.py:17]: Are you able to add a simple description of this argument in the docstring under the ``title`` description?

E.g. :param aspect: blah blah

**Review by robbievanleeuwen — COMMENTED** [review]
Looks good, thanks for this.

Only comments are the docstring for the ``plotting_context`` and black linting.

I'll have a look at how this generates all the plots in the docs to make sure everything looks ok when I'm at home tonight.

Cheers!

File: concreteproperties/post.py (hunk 1 above)
Comment (Agent6-6-6) [reply concreteproperties/post.py:17]: @robbievanleeuwen, should be resolved now provided the plots in docs are ok.

**Comment by robbievanleeuwen** [comment]
Thanks for this @Agent6-6-6!

**Pull request merged**